packages/syft/src/syft/service/user/user_service.py

```diff
@@ -84,6 +91,207 @@ def create(
+        result = method(context=root_context, notification=message)
+        if isinstance(result, SyftError):
+            return result
+
+        return SyftSuccess(message=success_msg)
+
+    @service_method(
+        path="user.request_password_reset",
+        name="request_password_reset",
+        roles=ADMIN_ROLE_LEVEL,
+    )
+    def request_password_reset(
+        self, context: AuthedServiceContext, uid: UID
+    ) -> str | SyftError:
+        result = self.stash.get_by_uid(credentials=context.credentials, uid=uid)
+        if result.is_err():
+            return SyftError(
+                message=(
+                    f"Failed to retrieve user with UID: {uid}. Error: {str(result.err())}"
+                )
+            )
+        user = result.ok()
+        if user is None:
+            return SyftError(message=f"No user exists for given: {uid}")
+
+        user_role = self.get_role_for_credentials(user.verify_key)
+        if user_role == ServiceRole.ADMIN:
+            return SyftError(
+                message="You can't request password reset for an Admin user."
+            )
+
+        user.reset_token = self.generate_new_password_reset_token(
+            context.server.settings.pwd_token_config
+        )
+        user.reset_token_date = datetime.now()
+
+        result = self.stash.update(
+            credentials=context.credentials, user=user, has_permission=True
+        )
+        if result.is_err():
+            return SyftError(
+                message=(
+                    f"Failed to update user with UID: {uid}. Error: {str(result.err())}"
+                )
+            )
+
+        return user.reset_token
+
+    @service_method(
+        path="user.reset_password", name="reset_password", roles=GUEST_ROLE_LEVEL
+    )
+    def reset_password(
+        self, context: AuthedServiceContext, token: str, new_password: str
+    ) -> SyftSuccess | SyftError:
+        """Resets a certain user password using a temporary token."""
+        result = self.stash.get_by_reset_token(
+            credentials=context.credentials, token=token
+        )
+        invalid_token_error = SyftError(
+            message=("Failed to reset user password. Token is invalid or expired!")
+        )
+
+        if result.is_err():
+            return SyftError(message="Failed to reset user password.")
+
+        user = result.ok()
+
+        # If token isn't found
+        if user is None:
+            return invalid_token_error
+
+        now = datetime.now()
+        time_difference = now - user.reset_token_date
+
+        # If token expired
+        expiration_time = context.server.settings.pwd_token_config.token_exp_min
+        if time_difference > timedelta(minutes=expiration_time):
+            return invalid_token_error
+
+        if not validate_password(new_password):
+            return SyftError(
+                message="Your new password must have at least 8 \
```

Comment:
Note: I think the `validate_password` does not check for these constraints. 

Also: Given that we generate the random password using Ascii and digits, there may be a possibility that random passwords don't follow these guidelines.

Reply:
We only generate temporary password reset tokens. The password itself is only set by the account owner.